### PR TITLE
chore: update PR title workflow to Conventional Commit style

### PR DESCRIPTION
**Pull request by patzick**
### 1. Why is this change necessary?
Introducing workflow change after the switch to GitHub

### 2. What does this change do, exactly?
Enforces Conventional Commit PR title, in order to keep PR history intact and also have a clean history after the merge to the head branch.

### 3. Describe each step to reproduce the issue or behaviour.
- 

### 4. Please link to the relevant issues (if any).
https://github.com/shopware/docs/pull/1620 - updated contribution guides

### 5. Checklist

- [ ] ~~I have rebased my changes to remove merge conflicts~~ - this removed
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [x] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfill them.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -21,7 +21,6 @@ Please create a second pull request at https://github.com/shopware/docs
 
 ### 5. Checklist
 
-- [ ] I have rebased my changes to remove merge conflicts
```
Comment (shyim) [inline .github/PULL_REQUEST_TEMPLATE.md:24]: why this is still valid?

File: .github/workflows/check-pr-title.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Check PR title
```
Comment (shyim) [inline .github/workflows/check-pr-title.yml:1]: instead introducing a new workflow, I would just adjust
https://github.com/shopware/shopware/blob/trunk/.danger.php#L230-L241

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (patzick) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: hey @shyim - I updated contribution docs, which is related to that line https://github.com/shopware/docs/pull/1620

even if we would change it to `i have resolved conflicts before creating PR` it still is a "living thing" so conflict may arise after PR is created and this checkbox becomes obsolete. In future there will be a stale bot/action to track stale and conflicted PRs so they are always up to date

Comment (aragon999) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: But when opening the PR the changes should still be rebased on the latest trunk, no?

Comment (patzick) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: should be `synced` with the head branch, and should stay `synced` at all times during the live of PR. Does not mean that needs to be `rebased`. Even better when you see PR history to spot when branch was created and at which parts of the work head branch was merged back to it

File: .github/workflows/check-pr-title.yml (hunk 2 above)
Comment (patzick) [reply .github/workflows/check-pr-title.yml:1]: it is a new workflow so check for that is very straightforward and descriptive. Lowers the barrier for newcomers to understand when needs to be improved. The `danger` script is mostly checking correctness of files, so I would leave it as a files linter. Similar workflows for the one introduces here are labelling or CLA check - related to PR process

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Phil23) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: ~~But a rebase is not prohibited, right? I am personally in favour of seeing the clean rebased history, to see which changes were added on top of the target branch. I don't see where the mixed history and merge commits adds benefits for the reviewer.~~

I just had a quick call with Patryk and he convinced me about the workflow. My biggest fear was, that the history looks crazy after a merge commit, which isn't actually the case. So I agree to the benefits the workflow brings in terms of reviewing just the stuff that was added over time to the PR. I guess it is important to highlight that we talk about everything after the PR was created, so collaboration can take place properly and that their is no need for a force push. Everything you do before you create your PR, like do a rebase to latest trunk, is still up to you.

**Review by keulinho — APPROVED** [review]

**Review by BrocksiNet — APPROVED** [review]
LGTM :+1:

**Pull request merged**